### PR TITLE
feat: add Nix flake

### PR DESCRIPTION
**Pull request by grialion**
Nix build uses GCC but provides clang-tools for development (devShell).

The /etc/lidm.ini is not kept after building though. You may want to add man documentation or make the nix derivation support it...

You can run the package using `nix run`:
```sh
nix run github:javalsai/lidm
```
Adding this to README.md might be useful.

### CODE REVIEW

**Comment by javalsai** [comment]
Completely missed the notification of opening this PR lol.
* Yeh, I thought of making man pages, not necessarily to talk about themes, but just as generic help, the binary doesn't support any type of flags including `-h`/`--help`, so some documentation would be nice. However, if the nix derivation could add it by default it'd be nice (I gotta look into it)
* I also have to add this to the readme or an installation guide.

**Review by javalsai — APPROVED** [review]
LGTM, just wanna see if I can configure the checks before merging

**Comment by javalsai** [comment]
image
yikes

**Comment by grialion** [comment]
> image yikes

Is LiDM or lidm the correct capitalization? It seems like I mixed them up... It builds bin/lidm but searches for bin/LiDM

**Pull request merged**
